autogluon/utils/tabular/ml/models/xgboost/callbacks.py

```diff
@@ -0,0 +1,167 @@
+
+    def callback(env):
+        """internal function"""
+        if not state:
+            init(env)
+        score = env.evaluation_result_list[-1][1]
+        best_score = state['best_score']
+        best_iteration = state['best_iteration']
+        maximize_score = state['maximize_score']
+        
+        if (maximize_score and score > best_score) or \
+                (not maximize_score and score < best_score):
+            msg = '[%d]\t%s' % (
+                env.iteration,
+                '\t'.join([_fmt_metric(x) for x in env.evaluation_result_list]))
+            state['best_msg'] = msg
+            state['best_score'] = score
+            state['best_iteration'] = env.iteration
+            # save the property to attributes, so they will occur in checkpoint.
+            if env.model is not None:
+                env.model.set_attr(best_score=str(state['best_score']),
+                                   best_iteration=str(state['best_iteration']),
+                                   best_msg=state['best_msg'])
+        elif env.iteration - best_iteration >= stopping_rounds:
+            best_msg = state['best_msg']
+            if verbose and env.rank == 0:
+                msg = "Stopping. Best iteration:\n{}\n\n"
+                rabit.tracker_print(msg.format(best_msg))
+            raise EarlyStopException(best_iteration)
+
+        if env.iteration % 10 == 0:
+            available = psutil.virtual_memory().available
+            cur_rss = mem_status.memory_info().rss
+            # rabit.tracker_print(f"{state}\n")
+            # rabit.tracker_print(f"CUR_RSS: {cur_rss}\n")
+            if cur_rss < state['init_mem_rss']:
+                state['init_mem_rss'] = cur_rss
+            estimated_model_size_mb = (cur_rss - state['init_mem_rss']) >> 20
+            available_mb = available >> 20
+
+            model_size_memory_ratio = estimated_model_size_mb / available_mb
+            if verbose and (model_size_memory_ratio > 0.25):
+                rabit.tracker_print(f'Available Memory: {available_mb} MB\n')
+                rabit.tracker_print(f'Estimated XGB model size: {estimated_model_size_mb} MB\n')
+
+            early_stop = False
+            if (model_size_memory_ratio > 1.0) or (available_mb < 512):
+                rabit.tracker_print('Warning: Large XGB model size may cause OOM error if training continues\n')
+                rabit.tracker_print(f'Available Memory: {available_mb} MB\n')
+                rabit.tracker_print(f'Estimated XGB model size: {estimated_model_size_mb} MB\n')
+                early_stop = True
+            
+            if early_stop:
+                if verbose and env.rank == 0:
+                    rabit.tracker_print(f'Warning: Early stopped GBM model prior to optimal result to avoid OOM error. Please increase available memory to avoid subpar model quality.\n')
+                    rabit.tracker_print(f'Early stopping. best iteration is:\n[{env.iteration}]\t{best_score}')
```

Comment:
shouldn't this be best_iteration instead of env.iteration?

Reply:
That's right. It should be `best_iteration`